### PR TITLE
Labx 15

### PR DESCRIPTION
**Pull request by Polycarpik**
@lastperson please review

### CODE REVIEW

File: contracts/StorageManager.sol [hunk 1]
```diff
@@ -0,0 +1,29 @@
+pragma solidity ^0.4.8;
+
+import './Owned.sol';
+
```
Comment (lastperson) [inline contracts/StorageManager.sol]: Single empty line is enough.

File: contracts/StorageManager.sol [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+contract StorageManager is Owned {
+
```
Comment (lastperson) [inline contracts/StorageManager.sol]: No need for this new line. Check other sol files for style guide.

File: contracts/StorageManager.sol [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	function blockAccess(address _actor, bytes32 _role) onlyContractOwner() returns(bool) {
+		approvedContracts[_actor][_role] = false;
+		AccessBlocked(_actor, _role);
+		return true;
+	}
+
+	function isAllowed(address _actor, bytes32 _role) constant returns(bool) {
+		return approvedContracts[_actor][_role];
+	}
+
```
Comment (lastperson) [inline contracts/StorageManager.sol]: No need for this new line. Check other sol files for style guide.

File: contracts/StorageManager.sol [hunk 4]
```diff
@@ -0,0 +1,29 @@
+pragma solidity ^0.4.8;
+
+import './Owned.sol';
+
+
+contract StorageManager is Owned {
+
+	mapping (address => mapping(bytes32 => bool)) internal approvedContracts;
```
Comment (lastperson) [inline contracts/StorageManager.sol]: sol files should have 4 spaces indentation. Here is tab used.

**Comment by Polycarpik** [comment]
Applied requested changes. @lastperson please review

File: contracts/StorageManager.sol [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    }
+    
+    function emitAccessGiven(address _user, bytes32 _role) {
+        AccessGiven(_user, _role, _getVersion());
+    }
+
+    function emitAccessBlocked(address _user, bytes32 _role) {
+        AccessBlocked(_user, _role, _getVersion());
+    }
+
```
Comment (lastperson) [inline contracts/StorageManager.sol]: Empty line can be removed.

File: contracts/StorageManager.sol [hunk 6]
```diff
@@ -0,0 +1,54 @@
+pragma solidity ^0.4.8;
+
+import './Owned.sol';
+import './EventsHistoryUser.sol';
+
+
```
Comment (lastperson) [inline contracts/StorageManager.sol]: One empty line can be removed.

File: contracts/StorageManager.sol [hunk 7]
```diff
@@ -0,0 +1,54 @@
+pragma solidity ^0.4.8;
```
Comment (lastperson) [inline contracts/StorageManager.sol]: We are using exact version (without ^ prefix) to not catch compiler bugs.

File: contracts/StorageManager.sol [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+import './Owned.sol';
+import './EventsHistoryUser.sol';
+
+
+contract StorageManager is EventsHistoryUser, Owned {
+
```
Comment (lastperson) [inline contracts/StorageManager.sol]: Empty line can be removed.

File: contracts/StorageManager.sol [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+	mapping (address => mapping(bytes32 => bool)) internal approvedContracts;
```
Comment (lastperson) [inline contracts/StorageManager.sol]: Tabs and spaces mixed together in this file. We are using 4 spaces indentation throughout all solidity files.

Comment (lastperson) [inline contracts/StorageManager.sol]: Should be `mapping(address => mapping(bytes32 => bool))` without space after `mapping`.
State variables are internal by default, no need to specify it explicitly.

File: contracts/StorageManager.sol [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+    function emitAccessGiven(address _user, bytes32 _role) {
+        AccessGiven(_user, _role, _getVersion());
+    }
+
+    function emitAccessBlocked(address _user, bytes32 _role) {
+        AccessBlocked(_user, _role, _getVersion());
+    }
+
+}
+
```
Comment (lastperson) [inline contracts/StorageManager.sol]: Empty line can be removed.

**Comment by Polycarpik** [comment]
Fixed, please merge @lastperson

**Pull request merged**